bdk_rpc_wallet_example/src/rpc.rs

```diff
@@ -0,0 +1,314 @@
+pub struct RpcClient {
+    client: Client,
+}
+
+impl Deref for RpcClient {
+    type Target = Client;
+    fn deref(&self) -> &Self::Target {
+        &self.client
+    }
+}
+
+impl RpcClient {
+    /// Initialize [`RpcClient`] from a [`RpcConfig`] and [`KeychainTxOutIndex`].
+    /// This creates a wallet inside bitcoin core, with a deterministic name
+    /// derived from the hash of descriptors within the [`KeychainTxOutIndex`].
+    /// Then imports all the spks from the index into the newly created core wallet.
+    pub fn init_for_tracker<K: Debug + Clone + Ord>(
+        config: &RpcConfig,
+        txout_index: &KeychainTxOutIndex<K>,
+    ) -> Result<Self, RpcError> {
+        let wallet_name = derive_name_from_keychainindex(txout_index);
+        let wallet_url = format!("{}/wallet/{}", config.url, wallet_name);
+        let client = Client::new(wallet_url.as_str(), config.auth.clone().into())?;
+        let rpc_version = client.version()?;
+        println!("rpc connection established. Core version : {}", rpc_version);
+        println!("connected to '{}' with auth: {:?}", wallet_url, config.auth);
+
+        if client.list_wallets()?.contains(&wallet_name.to_string()) {
+            println!("wallet already loaded: {}", wallet_name);
+        } else if list_wallet_dirs(&client)?.contains(&wallet_name.to_string()) {
+            println!("wallet wasn't loaded. Loading wallet : {}", wallet_name);
+            client.load_wallet(&wallet_name)?;
+            println!("wallet loaded: {}", wallet_name);
+        } else {
+            // pre-0.21 use legacy wallets
+            if rpc_version < 210_000 {
+                client.create_wallet(&wallet_name, Some(true), None, None, None)?;
+            } else {
+                // TODO: move back to api call when https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/225 is closed
+                let args = [
+                    Value::String(String::from(&wallet_name)),
+                    Value::Bool(true),  // disable_private_keys
+                    Value::Bool(false), //blank
+                    Value::Null,        // passphrase
+                    Value::Bool(false), // avoid reuse
+                    Value::Bool(false), // descriptor
+                ];
+                let _: Value = client.call("createwallet", &args)?;
+            }
+
+            println!("wallet created: {}", wallet_name);
+        }
+
+        let blockchain_info = client.get_blockchain_info()?;
+        let network = match blockchain_info.chain.as_str() {
+            "main" => Network::Bitcoin,
+            "test" => Network::Testnet,
+            "regtest" => Network::Regtest,
+            "signet" => Network::Signet,
+            _ => return Err(RpcError::General("Invalid Network string".to_string())),
+        };
+        if network != config.network {
+            return Err(RpcError::General("Invalid Network".to_string()));
+        }
```

Comment:
Do we need to do this? I would rather leave this for the caller to check that they are on the network they think they are on.